File: app/src/main/java/ru/henridellal/dialer/NumpadButton.java
package ru.henridellal.dialer;

import android.app.Activity;
import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.RelativeLayout;
import android.widget.TextView;

public class NumpadButton extends RelativeLayout {
	
	private String mChar;
	private String mLetters, mLocalLetters;
	
	public NumpadButton(Context context, AttributeSet attrs) {
		super(context, attrs);
		initButton(attrs);
	}
	
	public NumpadButton(Context context, AttributeSet attrs, int defStyleAttr) {
		super(context, attrs, defStyleAttr);
		initButton(attrs);
	}
	
	public NumpadButton(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
		super(context, attrs, defStyleAttr, defStyleRes);
		initButton(attrs);
	}
	
	private void initButton(AttributeSet attrs) {
		LayoutInflater inflater;
		if (getContext() instanceof Activity) {
			inflater = ((Activity)getContext()).getLayoutInflater();
		} else {
			inflater = LayoutInflater.from(getContext());
		}
		
		inflater.inflate(R.layout.numpad_button, this, true);
		TypedArray arr = getContext().obtainStyledAttributes(attrs, R.styleable.NumpadButton, 0, 0);
		try {
			mChar = arr.getString(R.styleable.NumpadButton_number);
			mLetters = arr.getString(R.styleable.NumpadButton_letters);
			mLocalLetters = arr.getString(R.styleable.NumpadButton_localLetters);
		} finally {
			arr.recycle();
		}
		((TextView)findViewById(R.id.numpad_number)).setText(mChar);
		((TextView)findViewById(R.id.numpad_letters)).setText(mLetters);
		setLetters(mLocalLetters);
	}

	public void setLetters(String letters) {
		mLocalLetters = letters;
		TextView numpadLettersLocalView = (TextView) findViewById(R.id.numpad_letters_local);
		if (null == mLocalLetters || mLocalLetters.isEmpty()) {
			numpadLettersLocalView.setVisibility(View.GONE);
		} else {
			numpadLettersLocalView.setVisibility(View.VISIBLE);
			numpadLettersLocalView.setText(mLocalLetters);
		}
	}
}

File: app/src/main/java/ru/henridellal/dialer/NumpadButton.kt
package ru.henridellal.dialer

import android.widget.RelativeLayout
import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.app.Activity
import ru.henridellal.dialer.R
import android.content.res.TypedArray
import android.view.View
import android.widget.TextView

class NumpadButton : RelativeLayout {
    private var mChar: String? = null
    private var mLetters: String? = null

    constructor(context: Context?, attrs: AttributeSet) : super(context, attrs) {
        initButton(attrs)
    }

    constructor(context: Context?, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        initButton(attrs)
    }

    constructor(context: Context?, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {
        initButton(attrs)
    }

    private fun initButton(attrs: AttributeSet) {
        val inflater: LayoutInflater
        inflater = if (context is Activity) {
            (context as Activity).layoutInflater
        } else {
            LayoutInflater.from(context)
        }
        inflater.inflate(R.layout.numpad_button, this, true)
        val arr = context.obtainStyledAttributes(attrs, R.styleable.NumpadButton, 0, 0)
        try {
            mChar = arr.getString(R.styleable.NumpadButton_number)
            mLetters = arr.getString(R.styleable.NumpadButton_letters)
        } finally {
            arr.recycle()
        }
        (findViewById<View>(R.id.numpad_number) as TextView).text = mChar
        (findViewById<View>(R.id.numpad_letters) as TextView).text = mLetters
    }

    fun setLetters(letters: String?) {
        mLetters = letters
        (findViewById<View>(R.id.numpad_letters) as TextView).text = mLetters
    }
}